### PR TITLE
Fix after debugging session

### PR DESCRIPTION
**Pull request by guerinoni**
- Arm: Fix start status of `cpsr`
- Thumb: Fix `uncond_branch`
- Interrupts: Add IE, IF, WAITCNT and one not used
- UI: Show register of CPU in hex

### CODE REVIEW

**Review by AlessioC31 — APPROVED** [review]

**Review by AlessioC31 — APPROVED** [review]

File: emu/src/render/ppu.rs [hunk 1]
```diff
@@ -62,15 +62,14 @@ impl PixelProcessUnit {
             }
             3 => {
                 // Bitmap mode
-                for y in 0..LCD_HEIGHT {
-                    for x in 0..LCD_WIDTH {
-                        let index_color = (y * LCD_WIDTH + x) * 2;
+                for x in 0..LCD_HEIGHT {
+                    for y in 0..LCD_WIDTH {
+                        let index_color = (x * LCD_WIDTH + y) * 2;
```
Comment (AlessioC31) [inline emu/src/render/ppu.rs:67]: I remember we had a discussion about this and @gallottino said that in the "graphics world" `y` is used for height and `x` for width

File: emu/src/render/ppu.rs [hunk 2]
```diff
@@ -81,22 +80,24 @@ impl PixelProcessUnit {
...
+                //         //     memory.video_ram[index_color],
+                //         //     memory.video_ram[index_color + 1],
+                //         // ]
+                //         // .into();
+                //         // FIXME
+                //         let color = Color(0);
+                //
+                //         gba_lcd.set_gbc_pixel(x, y, color);
+                //     }
+                // }
```
Comment (AlessioC31) [inline emu/src/render/ppu.rs:100]: Why commented?

File: emu/src/memory/internal_memory.rs [hunk 3]
```diff
@@ -110,7 +110,7 @@ impl IoDevice for InternalMemory {
             0x04000200..=0x040003FE => self.interrupts.write_at(address, value),
             0x05000000..=0x050001FF => self.bg_palette_ram[address - 0x05000000] = value,
             0x05000200..=0x050003FF => self.obj_palette_ram[address - 0x05000200] = value,
-            0x06000000..=0x06017FFF => self.video_ram[address - 0x06000000] = value,
+            0x06000000..=0x06FFFFFF => self.video_ram[address - 0x06000000] = value,
```
Comment (AlessioC31) [inline emu/src/memory/internal_memory.rs:113]: VRAM is ` 06000000-06017FFF   VRAM - Video RAM          (96 KBytes)` no?

File: emu/src/render/ppu.rs (hunk 1 above)
Comment (guerinoni) [reply emu/src/render/ppu.rs:67]: Mmmm can be true, but is only temporary, this will be probably changed :)

File: emu/src/render/ppu.rs (hunk 2 above)
Comment (guerinoni) [reply emu/src/render/ppu.rs:100]: Right now we are not using this mode, just for being sure of what we need time to time

File: emu/src/memory/internal_memory.rs (hunk 3 above)
Comment (guerinoni) [reply emu/src/memory/internal_memory.rs:113]: Yes, I extended to also NotUsed part because I want to avoid other problem of read/write on non used memory

Comment (AlessioC31) [reply emu/src/memory/internal_memory.rs:113]: I think it would be better to put not_used explicitly instead of increasing the vram

**Review by AlessioC31 — APPROVED** [review]

**Review by AlessioC31 — APPROVED** [review]

**Pull request merged**